src/datasources/cache/cache.first.data.source.spec.ts

```diff
@@ -35,11 +36,78 @@ describe('CacheFirstDataSource', () => {
     );
   });
 
-  it('should return the data returned by the underlying network interface', async () => {
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+
+  it('should return the data returned by the underlying network interface and cache it', async () => {
+    const targetUrl = faker.internet.url({ appendSlash: false });
+    const cacheDir = new CacheDir(faker.word.sample(), faker.word.sample());
+    const notFoundExpireTimeSeconds = faker.number.int();
+    const data = JSON.parse(fakeJson());
+    mockNetworkService.get.mockImplementation((url) => {
+      switch (url) {
+        case targetUrl:
+          return Promise.resolve({ data });
+        default:
+          return Promise.reject(`No matching rule for url: ${url}`);
+      }
+    });
+
+    const actual = await cacheFirstDataSource.get({
+      cacheDir,
+      url: targetUrl,
+      notFoundExpireTimeSeconds,
+    });
+
+    expect(actual).toEqual(data);
+    expect(mockNetworkService.get).toHaveBeenCalledTimes(1);
+    expect(fakeCacheService.keyCount()).toBe(1); // only data is cached (as no invalidation happened yet a this point in time)
+    expect(await fakeCacheService.get(cacheDir)).toEqual(JSON.stringify(data));
```

Comment:
nitpick: mostly syntactic sugar that we have been using in other tests.

```suggestion
    await expect(fakeCacheService.get(cacheDir)).resolves.toEqual(
      JSON.stringify(data),
    );
```